infra/blockchain/evm.go

```diff
@@ -49,15 +53,18 @@ func AddChainClient(chain models.Chain) {
 		Client:  client,
 		ChainId: chain.Id,
 	}
+	logrus.WithFields(logrus.Fields{
+		"id": chain.Id, "name": chain.Name, "rpc": chain.Rpc,
+	}).Info("chain added")
 }
 func AddCrossRequest(chainId int64, txHash string) (*models.CrossRequest, error) {
 	evm, ok := clients[chainId]
 	if !ok {
 		return nil, fmt.Errorf("chain not found: %v", chainId)
 	}
 
-	if _, _, err := evm.Client.TransactionByHash(context.Background(), common.HexToHash(txHash)); err != nil {
-		return nil, fmt.Errorf("TransactionByHash error: [%v], [%v]", txHash, err.Error())
+	if _, _, err := evm.TransactionByHash(txHash); err != nil {
```

Comment:
shall we need to split conditions between txn not found and internal server error?